app.py

```diff
@@ -17,99 +17,17 @@
 )
 
 
-def get_authorization_header(root, discharge):
-    """
-    Bind root and discharge macaroons and return the authorization header.
-    """
-
-    bound = Macaroon.deserialize(root).prepare_for_request(
-        Macaroon.deserialize(discharge)
-    )
-
-    return 'Macaroon root={}, discharge={}'.format(root, bound.serialize())
-
-
-def is_authenticated():
-    return (
-        'openid' in flask.session and
-        'macaroon_discharge' in flask.session and
-        'macaroon_root' in flask.session
-    )
-
-
-def empty_session():
-    flask.session.pop('macaroon_root', None)
-    flask.session.pop('macaroon_discharge', None)
-    flask.session.pop('openid', None)
-
-
 def redirect_to_login():
     return flask.redirect(''.join([
         'login?next=',
         flask.request.url_rule.rule,
     ]))
 
 
-def request_macaroon():
-    response = requests.request(
-        url='https://dashboard.snapcraft.io/dev/api/acl/',
-        method='POST',
-        json={'permissions': ['package_access']},
-        headers={
-            'Accept': 'application/json, application/hal+json',
-            'Content-Type': 'application/json',
-            'Cache-Control': 'no-cache',
-        }
-    )
-
-    return response.json()['macaroon']
-
-
-def verify_macaroon(root, discharge, url):
-    authorization = get_authorization_header(root, discharge)
-    response = requests.request(
-        url='https://dashboard.snapcraft.io/dev/api/acl/verify/',
-        method='POST',
-        json={
-            'auth_data': {
-                'authorization': authorization,
-                'http_uri': url,
-                'http_method': 'GET'
-            }
-        },
-        headers={
-            'Accept': 'application/json, application/hal+json',
-            'Content-Type': 'application/json',
-            'Cache-Control': 'no-cache',
-        }
-    )
-
-    return response.json()
-
-
-def get_refreshed_discharge(discharge):
-    url = (
-        'https://login.ubuntu.com'
-        '/api/v2/tokens/refresh'
-    )
-    response = requests.request(
-        url=url,
-        method='POST',
-        json={'discharge_macaroon': discharge},
-        headers={
-            'Accept': 'application/json, application/hal+json',
-            'Content-Type': 'application/json',
-            'Cache-Control': 'no-cache',
-        }
-    )
-
-    return response.json()['discharge_macaroon']
-
-
 @app.route('/')
 def homepage():
     context = {}
-    if is_authenticated():
+    if authentication.is_authenticated(flask.session):
         context['connected'] = True
```

Comment:
Not sure if this part is necessary (apart from debugging), and 'connected' name may be confusing.

Reply:
This is not the best, but it's not something that is going to be copied past to other projects.

Reply:
Yeah it's not ideal but it's not part of this PR or the library code.